logstash-core/src/main/java/org/logstash/Logstash.java

```diff
@@ -0,0 +1,144 @@
+package org.logstash;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jruby.Ruby;
+import org.jruby.RubyException;
+import org.jruby.RubyInstanceConfig;
+import org.jruby.RubyNumeric;
+import org.jruby.exceptions.RaiseException;
+import org.jruby.runtime.builtin.IRubyObject;
+
+/**
+ * Logstash Main Entrypoint.
+ */
+public final class Logstash implements Runnable, AutoCloseable {
+
+    private static final Logger LOGGER = LogManager.getLogger(Logstash.class);
+
+    /**
+     * Configuration for {@link #ruby}.
+     */
+    private final RubyInstanceConfig config;
+
+    /**
+     * JRuby Runtime Environment.
+     */
+    private final Ruby ruby;
+
+    /**
+     * Ruby Entrypoint Script.
+     */
+    private final InputStream script;
+
+    /**
+     * Main Entrypoint.
+     * Requires environment {@code "LS_HOME"} to be set to the Logstash root directory.
+     * @param args Logstash CLI Arguments
+     */
+    public static void main(final String... args) {
+        final String lsHome = System.getenv("LS_HOME");
+        if (lsHome == null) {
+            throw new IllegalStateException("LS_HOME environment variable must be set.");
```

Comment:
Can we make this error message more actionable? If a user sees this exception, what should they do to resolve it, and can we include hints for resolving this in the message?

Reply:
@jordansissel hmm I think the user can't do anything they see this message (other than open an issue here). This being thrown means there is a bug in the `bin/logstash` or `bin/logstash.bat`. I simply put this check here for our convenience to have a clean failure instead of some other failure with a strange `null` in some string.

Reply:
> other than open an issue here

Ok cool. Let's tell them it's probably a bug in the error message.

Reply:
@jordansissel isn't probably a bug implied in this being a `RuntimeException`? :)

Reply:
As a user, I don't personally associate "this is a bug" with RuntimeException. I'd rather be explicit.  Saying "must be set" implies to me that the user should take action to set LS_HOME, which likely isn't the case. Otherwise, a user will end up struggling with this action trying to figure out how, and what value, to set LS_HOME.